src/Common/FitOnFhir.Common/Services/AuthStateService.cs

```diff
@@ -0,0 +1,85 @@
+
+namespace Microsoft.Health.FitOnFhir.Common.Services
+{
+    public class AuthStateService : IAuthStateService
+    {
+        private readonly AuthenticationConfiguration _authenticationConfiguration;
+        private readonly IJwtSecurityTokenHandlerProvider _jwtSecurityTokenHandlerProvider;
+        private readonly ILogger _logger;
+
+        public AuthStateService(
+            AuthenticationConfiguration authenticationConfiguration,
+            IJwtSecurityTokenHandlerProvider jwtSecurityTokenHandlerProvider,
+            ILogger<AuthStateService> logger)
+        {
+            _authenticationConfiguration = EnsureArg.IsNotNull(authenticationConfiguration, nameof(authenticationConfiguration));
+            _jwtSecurityTokenHandlerProvider = EnsureArg.IsNotNull(jwtSecurityTokenHandlerProvider, nameof(jwtSecurityTokenHandlerProvider));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public string ExternalIdentifier { get; set; }
+
+        public string ExternalSystem { get; set; }
+
+        /// <inheritdoc/>
+        public AuthState CreateAuthState(HttpRequest httpRequest)
+        {
+            var externalId = HttpUtility.UrlDecode(httpRequest.Query[Constants.ExternalIdQueryParameter]);
+
+            var externalSystem = HttpUtility.UrlDecode(httpRequest.Query[Constants.ExternalSystemQueryParameter]);
+
+            // is this for anonymous logins?
+            if (_authenticationConfiguration.IsAnonymousLoginEnabled)
+            {
+                ExternalIdentifier = EnsureArg.IsNotNullOrWhiteSpace(externalId, nameof(externalId));
+                ExternalSystem = EnsureArg.IsNotNullOrWhiteSpace(externalSystem, nameof(externalSystem));
+            }
+            else
+            {
+                // do not allow the ExternalId or ExternalSystem query params when authentication is enabled
+                if (!string.IsNullOrEmpty(externalId) || !string.IsNullOrEmpty(externalSystem))
+                {
+                    _logger.LogError($"{Constants.ExternalIdQueryParameter} and {Constants.ExternalSystemQueryParameter} are forbidden query parameters with non-anonymous authorization.");
+                    throw new ArgumentException();
+                }
+
+                // extract the token from the header.
+                if (!httpRequest.TryGetTokenStringFromAuthorizationHeader(JwtBearerDefaults.AuthenticationScheme, out string token))
+                {
+                    _logger.LogError("The request Authorization header is invalid.");
+                    return default;
+                }
+
+                // Read the token
+                _jwtSecurityTokenHandlerProvider.SetMapInboundClaims(false);
+                var jwtSecurityToken = _jwtSecurityTokenHandlerProvider.ReadJwtToken(token);
+
+                if (jwtSecurityToken == default)
+                {
+                    _logger.LogError("The JWT token is empty.");
```

Comment:
nit: I don't think you need to log this the `_jwtSecurityTokenHandlerProvider` already logs the error.